orchestrator/ingest/ingest.go

```diff
@@ -0,0 +1,186 @@
+package ingest
+
+import (
+	"fmt"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/SayedAlesawy/Videra-Ingestion/orchestrator/config"
+	"github.com/SayedAlesawy/Videra-Ingestion/orchestrator/drivers/tcp"
+	"github.com/SayedAlesawy/Videra-Ingestion/orchestrator/utils/errors"
+	"github.com/SayedAlesawy/Videra-Ingestion/orchestrator/utils/params"
+	"github.com/pebbe/zmq4"
+)
+
+// logPrefix Used for hierarchical logging
+var logPrefix = "[Ingestion-Manager]"
+
+// ingestionManagerOnce Used to garauntee thread safety for singleton instances
+var ingestionManagerOnce sync.Once
+
+// ingestionManagerInstance A singleton instance of the ingestion manager object
+var ingestionManagerInstance *IngestionManager
+
+// ack Ack sent by worker pool
+const ack = "ack"
+
+// IngestionManagerInstance A function to return an ingestion manager instance
+func IngestionManagerInstance() *IngestionManager {
+	ingestionManagerOnce.Do(func() {
+		configManager := config.ConfigurationManagerInstance("config/config_files")
+		configObj := configManager.IngestionManagerConfig("ingestion_manager.yaml")
+		params := params.OrchestratorParamsInstance()
+
+		connection, err := tcp.NewConnection(zmq4.REQ, "")
+		errors.HandleError(err, fmt.Sprintf("%s %s\n", logPrefix, "Unable to establish tcp connection"), true)
+
+		activeRoutines := 1
```

Comment:
if it's a constant then move it outside of the function block, same as ack

Reply:
Actually it's not a constant that would be used outside. I just couldn't find a way to count how many go routines I spawn, so I add it manually. The variable is only used in the constructor so I don't think it needs to be a constant. Later it's used as `manager.activeRoutines`